[Er Wen Han] iP

seanflyyy opened the pull request:
# Froggy

> "When something is important enough, you do it even if the odds are not in your favour." – Elon Musk ([source](https://www.brainyquote.com/authors/elon-musk-quotes))

Froggy helps you achieve the above goal by consolidating the things you need to do and decide what is the most important. It is
- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is:
1. download it from [here](https://github.com/seanflyyy/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
And it is **FREE**!

Features:

- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
If you Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

seanflyyy committed:
add Greet, Echo, Exit

seanflyyy committed:
add Add, List functionality

seanflyyy committed:
add Task class; mark, unmark functionality for task

seanflyyy committed:
update access modifier from private to public for getStatusIcon() function

seanflyyy committed:
add 3 classes - Deadline, Event and ToDo which extend the Task class

seanflyyy committed:
add logic in Duke.java to handle creation of the 3 different kinds of tasks, ToDo, Event and Deadline

seanflyyy committed:
add A-TextUiTesting

seanflyyy committed:
add handling of errors

seanflyyy committed:
add delete logic and refactored mark and unmark logic

seanflyyy committed:
refactor code to split the command at the very beginning when the command is read

seanflyyy committed:
Implement Level-7

This commit will add FileManager class so as to handle the read, create and writing functionality of the file.

It has also made some minor modifications to the Task, ToDo, Deadline and Event class, adding the toFileString()
method, which prints out the data in a format to be stored in the .txt file.

The Duke class has also been modified to incorporate the writeToFile functionality whenever there are changes.

seanflyyy committed:
Implement Level-8

Modify Deadline and Event class to now accept dateTime as String and store it as a LocalDate.

seanflyyy committed:
Merge branch 'branch-Level-8'

seanflyyy committed:
Implement A-MoreOOP

Add AddCommand, DeleteCommand, ExitCommand, InvalidCommand, ListCommand, MarkCommand that all extend the Command class.

Add Parser, Storage, TaskList, Ui classes.

Refactored most of the code to make it more OOP

seanflyyy committed:
Implement A-Packages

Add minor bug fix which fixes.

Refactor code to package java classes into their own respective packages.

seanflyyy committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

seanflyyy committed:
Implement A-Packages (previous commit did not place the files in the right package)

seanflyyy committed:
Implement A-Gradle

Able to build and run duke using gradle

seanflyyy committed:
Implement A-JUnit

Add test methods for DeadlineTest and TaskListTest

seanflyyy committed:
Implement A-JavaDoc

seanflyyy committed:
Add updated input, runtest.bat and .gitignore

seanflyyy committed:
Implement A-CodingStandard

seanflyyy committed:
Implement Level-9

Add functionality to search for a keyword in all of the tasks under TaskList.

seanflyyy committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Implement A-CodingStandard

# Conflicts:
#	src/main/java/seedu/duke/parser/Parser.java
#	src/main/java/seedu/duke/storage/Storage.java
#	src/main/java/seedu/duke/task/TaskList.java

seanflyyy committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Implement Level-9

# Conflicts:
#	src/main/java/seedu/duke/parser/Parser.java
#	text-ui-test/input.txt

seanflyyy committed:
Fix JavaDoc CheckStyle issues

LJXSean commented on src/main/java/seedu/duke/storage/Storage.java:
```diff
@@ -0,0 +1,111 @@
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
```
Perhaps a line break would make it easier to browse the import statements list?

seanflyyy replied:
Thank you! I've made the changes

LJXSean commented on src/main/java/seedu/duke/ui/Ui.java:
```diff
@@ -0,0 +1,143 @@
+        System.out.println(Style.INDENTATION + "Nice! I've marked this seedu.duke.task as done:");
+        System.out.println(Style.INDENTATION + Style.HALF_INDENTATION + task + "\n");
+    }
+
```
Did you intend to add another empty line here? Perhaps one would look cleaner instead.

LJXSean commented on src/test/java/seedu/duke/task/DeadlineTest.java:
```diff
@@ -0,0 +1,21 @@
+
+class DeadlineTest {
+    @Test
+    public void deadlineConstructorNoIsDone_normalDeadlineWithDeadlineDateTime_toStringCorrect() {
```
I like that your test names are descriptive 👍

LJXSean commented on src/main/java/seedu/duke/storage/Storage.java:
```diff
@@ -0,0 +1,111 @@
+        this.filePath = filePath;
+    }
+
+
```
Noticed the double empty lines here too!

seanflyyy replied:
Great catch! Thank you!

LJXSean reviewed (commented):
LGTM, other than some minor nitpicks! 👍

seanflyyy committed:
Modify the code slightly, fix some formatting and display issues

seanflyyy committed:
Implement Level-10

Add a GUI to Froggy, allowing for interaction and commands to be processed through a UI

Hoodineee commented on src/main/java/seedu/duke/MainWindow.java:
```diff
@@ -0,0 +1,51 @@
+
+    private Duke duke;
+
+    private Image userImage = new Image(this.getClass().getResourceAsStream("/images/dragon.png"));
+    private Image dukeImage = new Image(this.getClass().getResourceAsStream("/images/froggy.png"));
```
I think these fields can be final since the images wont be changing.

seanflyyy replied:
Thanks for the tip!

Hoodineee commented on src/main/java/seedu/duke/task/Task.java:
```diff
@@ -0,0 +1,72 @@
+    /**
+     * Marks the task as done.
+     */
+    public void markAsDone() {
```
I like the name of this method!

Hoodineee commented on src/main/java/seedu/duke/task/TaskList.java:
```diff
@@ -0,0 +1,127 @@
+     *
+     * @param keyword Specifies the keyword to match the tasks to.
+     */
+    public String displayMatchingTasks(String keyword) {
```
I like this method name!

Hoodineee commented on src/main/java/seedu/duke/ui/Ui.java:
```diff
@@ -0,0 +1,166 @@
+    }
+
+    /**
+     * Displays teh task that has been added to ArrayList.
```
I think there is a misspelling here.

seanflyyy replied:
Good catch! Thank you!

Hoodineee reviewed (commented):
Really clean and good code! The issues I found are really minor and doesn't affect the functionality of the code. LGTM

seanflyyy committed:
Resolve PR reviews

seanflyyy committed:
Implement A-Assertions

Add assertions to ensure that the number of Tasks is greater than 0.
If not, it should not be able to display the number of tasks.

Fix some UI features to do with the indenting so as to allow for a
cleaner response message from the bot.

seanflyyy committed:
Improve CodeQuality

Improve code quality by making the happy path more prominent as consequence,
avoiding the arrowhead style code.

Add taskLabel field to each task class (Deadline, Event and Todo) to reduce reliance
on magic literals.

SLAPped hard in the case of AddCommand.java, Storage.java, extracting a method out so as
to ensure SLAP.

seanflyyy committed:
Add more assertions

seanflyyy committed:
Merge pull request #2 from seanflyyy/branch-A-CodeQuality

Improve CodeQuality

seanflyyy committed:
Merge branch 'master' into branch-A-Assertions

seanflyyy committed:
Merge pull request #1 from seanflyyy/branch-A-Assertions

Branch a assertions

seanflyyy committed:
Set up CI

seanflyyy committed:
Implement C-Update

Create the update functionality so that a user can now update a task's description or a task's date and time.

seanflyyy committed:
Rename comments and variables

seanflyyy committed:
Add files via upload

seanflyyy committed:
Update README.md

seanflyyy committed:
Update README.md

seanflyyy committed:
Update the project structure, removing the seedu directory

seanflyyy committed:
Update the code to now include the froggy package

seanflyyy committed:
Update User Guide

seanflyyy committed:
Merge branch 'master' of https://github.com/seanflyyy/ip

* 'master' of https://github.com/seanflyyy/ip:
  Update README.md
  Update README.md
  Add files via upload

seanflyyy committed:
Update README.md

seanflyyy committed:
Delete Ui.png

seanflyyy committed:
Add files via upload

seanflyyy committed:
Update README.md

seanflyyy committed:
Update the user guide with relevant screenshots

seanflyyy committed:
Update build.gradle file to be the most updated mainClassName

seanflyyy committed:
Update update command handling and add closing of the application feature upon typing Bye command

seanflyyy committed:
Refactor file structure
